bin/inference/pycbc_inference_plot_samples

```diff
@@ -44,44 +52,87 @@ opts = parser.parse_args()
 # setup log
 pycbc.init_logging(opts.verbose)
 
+rc('text', usetex=True)
+
 # load the results
 fp, parameters, labels, _ = io.results_from_cli(opts,
-                                                load_samples=False)
+                                                load_samples=False,
+                                                walkers=None)
 
 # get number of dimensions
 ndim = len(parameters)
 
+# get walker indices
+if opts.walkers == ['all'] or opts.walkers == None:
+    walkers = range(fp.nwalkers)
+else:
+    walkers = list(map(int, opts.walkers))
+
 # plot samples
 # plot each parameter as a different subplot
 logging.info("Plotting samples")
-fig, axs = plt.subplots(ndim, sharex=True)
+fig, axs = plt.subplots(ndim, figsize=(10,8), sharex=True)
 plt.xlabel("Iteration")
 
 # loop over parameters
 axs = [axs] if not hasattr(axs, "__iter__") else axs
+
+file_parameters, cs = transforms.get_common_cbc_transforms(
+                                            parameters, fp.variable_params)
+if opts.thin_start is not None:
+    xmin = opts.thin_start
+else:
+    xmin = fp.attrs['thin_start']
+
+if opts.thin_end is not None:
+    xmax = opts.thin_end
+else:
+    try:
+        xmax = fp.attrs['thin_end']
+    except KeyError:
+        xmax = fp.niterations
+
+if opts.thin_interval is not None:
+    xint = opts.thin_interval
+else:
+    xint = fp.attrs['thin_interval']
+
+if xmin > xmax:
+    raise ValueError("Start iteration {} provided for reading samples is "
+                     "higher than the end iteration {}.".format(xmin, xmax))
+xvals = numpy.arange(xmin, xmax, xint)
+
 for i, arg in enumerate(parameters):
-    # plot each walker as a different line on the subplot
-    for j in range(fp.nwalkers):
-        # plot each walker as a different line on the subplot
-        file_parameters, cs = transforms.get_common_cbc_transforms(
-                                                 parameters, fp.variable_params)
-        y = fp.read_samples(file_parameters, walkers=j,
-                            thin_start=opts.thin_start,
-                            thin_interval=opts.thin_interval,
-                            thin_end=opts.thin_end)
-        y = transforms.apply_transforms(y, cs)
-
-        axs[i].plot(y[arg], alpha=0.25)
-        # Set y labels
-        axs[i].set_ylabel(labels[arg])
+    chains_arg = []
+    for widx in walkers:
+        chain = fp.read_samples(file_parameters, temps=opts.temps,
+                                walkers=widx, thin_start=opts.thin_start,
+                                thin_interval=opts.thin_interval,
+                                thin_end=opts.thin_end)
+        chain = transforms.apply_transforms(chain, cs)
+        chains_arg.append(chain[arg])
+    if opts.walkers is not None:
+        for chain in chains_arg:
+            # plot each walker as a different line on the subplot
+            axs[i].plot(xvals, chain, alpha=0.6)
```

Comment:
Following up on my comment above, here, instead of doing `xvals`, do:
```
axs[i].plot((numpy.arange(len(chain)+xmin)*thinned_by, chain, alpha=0.6)
```